src/contextify.cc

```diff
@@ -180,170 +182,201 @@ class ContextWrap : public ObjectWrap {
         return true;
     }
 
-    static NAN_PROPERTY_GETTER(GlobalPropertyGetter) {
-        NanScope();
-        Local<Object> data = args.Data()->ToObject();
+    static void GlobalPropertyGetter(Local<String> property, const PropertyCallbackInfo<Value>& info) {
+        Local<Object> data = info.Data()->ToObject();
         ContextifyContext* ctx = ObjectWrap::Unwrap<ContextWrap>(data)->ctx;
-        if (!ctx)
-            NanReturnUndefined();
-        Local<Value> rv = NanNew(ctx->sandbox)->GetRealNamedProperty(property);
-        if (rv.IsEmpty()) {
-            rv = NanNew(ctx->proxyGlobal)->GetRealNamedProperty(property);
+
+        if (!ctx) {
+            info.GetReturnValue().SetUndefined();
+            return;
         }
-        NanReturnValue(rv);
+
+        Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
+
+//        if (rv.IsEmpty()) {
+//            rv = Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property);
+//        }
+
+        info.GetReturnValue().Set(rv);
     }
 
-    static NAN_PROPERTY_SETTER(GlobalPropertySetter) {
-        NanScope();
-        Local<Object> data = args.Data()->ToObject();
+    static void GlobalPropertySetter(Local<String> property, Local<Value> value, const PropertyCallbackInfo<Value>& info) {
```

Comment:
It's a wrapper function to normalize the callback signature. Performance differences should be negligible, since they drown in the overhead of js -> c++ transition. The same happens for NAN_METHOD and all the others. If your accessors or whatever would get called, they get called. You must, however use the right function to register said callbacks, e.g. Nan::SetAccessor

On November 3, 2015 10:32:46 AM GMT+02:00, Ido Ben-Yair <notifications@github.com> wrote:
>>      }
>>  
>> -    static NAN_PROPERTY_SETTER(GlobalPropertySetter) {
>> -        NanScope();
>> -        Local<Object> data = args.Data()->ToObject();
>> +    static void GlobalPropertySetter(Local<String> property,
>Local<Value> value, const PropertyCallbackInfo<Value>& info) {
>
>NAN did something weird involving overriding my callback with its own
>and, performance aside, I wouldn't have minded, but it ended up not
>calling my function at all. So I had to ditch it.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/brianmcd/contextify/pull/190/files#r43724488